packages/paste-core/components/file-picker/src/index.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {Label, RequiredDot} from '@twilio-paste/label';
+import {Text} from '@twilio-paste/text';
+import {useUID} from '@twilio-paste/uid-library';
+import {Button} from '@twilio-paste/button';
+import {MediaObject, MediaBody, MediaFigure} from '@twilio-paste/media-object';
+import {Box} from '@twilio-paste/box';
+import type {BoxProps} from '@twilio-paste/box';
+
+export interface FilePickerProps extends React.HTMLAttributes<HTMLInputElement>, Pick<BoxProps, 'element'> {
+  disabled?: boolean;
+  required?: boolean;
+}
+
+const FilePicker = React.forwardRef<HTMLInputElement, FilePickerProps>(
+  ({element = 'FILEPICKER_CONTAINER', disabled = false, required = false, ...props}, ref) => {
+    const [fileName, setFileName] = React.useState('No files selected');
+
+    const id = useUID();
+
+    const handleChange = (evt: any): void => {
+      const file = evt.target.value;
+      setFileName(file);
+      console.log('the event', evt);
+      console.log('file name', fileName);
+    };
+
+    return (
+      <MediaObject {...props}>
+        <MediaFigure spacing="space30">
+          <Label htmlFor={id} element={element}>
+            <Box
+              // The actual <input type="file"/> is hidden but still appears in the DOM
+              id={id}
+              ref={ref}
+              as="input"
+              type="file"
+              aria-disabled={disabled}
+              size="size0"
+              border="none"
+              overflow="hidden"
+              padding="space0"
+              margin="space0"
+              whiteSpace="nowrap"
+              textTransform="none"
+              position="absolute"
+              clip="rect(0 0 0 0)"
+              onChange={handleChange}
+            />
+            {required ? <RequiredDot /> : <></>}
```

Comment:
suggestion: I'd be tempted to just leave the Label component to do the display of the required dot, as it does it for us. You do still want to be able to place the `required` HTML attribute on the input element though, so keep that in mind too.

```
<Label htmlFor="resume" required>Resume</Label>
<FilePicker aria-describedby="resume_help_text" id="resume" name="resume" onChange="" required/>
<HelpText id="resume_help_text">Use your work email address.</HelpText>
```